### ISSUE TITLE
Callback URL generated with http not https, despite by-the-book config

### ISSUE THREAD

**Issue by bradspry** [lyrasis/aspace-oauth#11]
Please see the protocol just after the 'service' variable below:

`https://webauth.uncc.edu/idp/profile/cas/login?service=http://archivesspace.uncc.edu/auth/cas/callback?url=https%3A%2F%2Farchivesspace.uncc.edu%2F`

```
AppConfig[:authentication_sources] = [{
model: 'ASOauth',
    provider: 'cas',
    label: 'CAS Sign In',
    slo_link: true,
    config: {
      url: 'https://webauth.uncc.edu',
      host: 'webauth.uncc.edu',
      login_url: '/idp/profile/cas/login',
      logout_url: '/idp/profile/cas/logout',
      service_validate_url: '/idp/profile/cas/serviceValidate'
      # if your server does not return an email address, you can add one
      # here using the fetch_raw_info option.
      # fetch_raw_info: ->(s, o, t, user_info) {  { email: "#{user_info['user']}@uncc.edu" } } 
} }]
```

**Comment by mark-cooper** [comment]
Hi @bradspry 

The config section is just passing through to the cas gem: https://github.com/dlindahl/omniauth-cas

That gem is a bit hard to follow. It looks like the callback_url section isn't what you are expecting, although the return url does have https. Going by the easiest possible thing to do have you tried adding `ssl: true` to your config? (I see it says optional in the docs but I'm not sure I believe it)

**Comment by bradspry** [comment]
Yes, I tested `ssl: true` many iterations and still no joy.  I also arrived at it being optional, so I eliminated it from the equation.

I also have these set based upon this conversation: https://github.com/lyrasis/aspace-oauth/issues/1

## Proxy URLs
AppConfig[:frontend_proxy_url] = "https://archivesspace.uncc.edu"
AppConfig[:public_proxy_url] = "https://findingaids.uncc.edu"

**Comment by mark-cooper** [comment]
Ah, "still no joy", we've all been there, although it's usually SAML for us =)

The CAS configs we have do follow in line with the example. One thing I see happening is that when I click the login link from aspace I'm taken to the IDP login screen and the params are correct, even without us explicitly defining the callback url. Digging a little I think this happening in the omniauth gem, and the scheme is resolved here:

https://github.com/omniauth/omniauth/blob/v1.7.1/lib/omniauth/strategy.rb#L506-L512

We use Nginx and set at least the `X-Forwarded-Proto` $scheme (https when https) which is how I think it's being set correctly.

**Comment by bradspry** [comment]
Thank you so much for your time.

I have read about a callback_url variable; is it possible to implicitly set in config: ?

**Comment by mark-cooper** [comment]
Although the cas gem docs seem to indicate it's possible I don't see that it is. On the omniauth gem side it looks like it accepts a value for `full_host` (`full_host: "https://archivesspace.uncc.edu"`), which would bypass the scheme determination based on request headers, but I'm not sure that will make it through from cas. ...

**Comment by bradspry** [comment]
It was first documented by @yanrubi in Issue #1:
https://github.com/lyrasis/aspace-oauth/issues/1#issuecomment-392918424

I wonder if @yanrubi ever figured it out?

If I can't figure it out, I'll be shifting gears to try aspace-oauth SAML.

**Comment by mark-cooper** [comment]
@bradspry are you able to update your web proxy config to include one of the headers that sets the scheme to https in the omniauth gem?

Here's an example for nginx: https://github.com/archivesspace/archivesspace/blob/master/proxy/default.conf#L32

I think doing that would resolve your issue.

**Comment by bradspry** [comment]
I will work on implementation. Thank you for pointing me in a good direction.  Much appreciated.

**Comment by bradspry** [comment]
I have round-trip communications happening now, thank you so much.

Some final detail remains on my ArchivesSpace CAS config to connect the final dot; getting an "authentication error, unable to login." message after the idP returns the session back to the AS server...

**Comment by bradspry** [comment]
I have CAS working now. I'll contribute what I learned to example CAS configuration in README.md
